src/rsa/keypair.rs

```diff
@@ -563,21 +569,48 @@ impl KeyPair {
         msg: &[u8],
         signature: &mut [u8],
     ) -> Result<(), error::Unspecified> {
-        let mod_bits = self.public.n().len_bits();
-        if signature.len() != mod_bits.as_usize_bytes_rounded_up() {
+        if signature.len() != self.public().modulus_len() {
             return Err(error::Unspecified);
         }
 
         let m_hash = digest::digest(padding_alg.digest_alg(), msg);
-        padding_alg.encode(m_hash, signature, mod_bits, rng)?;
+
+        // Use the output buffer as the scratch space for the signature to
+        // reduce the required stack space.
+        padding_alg.encode(m_hash, signature, self.public().n().len_bits(), rng)?;
+
+        // RFC 8017 Section 5.1.2: RSADP, using the Chinese Remainder Theorem
+        // with Garner's algorithm.
+
+        // Steps 1 and 2.
+        let m = self.private_exponentiate(signature)?;
+
+        // Step 3.
```

Comment:
Should this still keep the reference to "Manger's Attack revisited"?

Reply:
I mentioned why I removed the reference in the third commit message.

Reply:
Ah ok, thank you for the clarification.